### PR TITLE
Extract `edge_gen`

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### What does this implement/fix? Explain your changes.
All graph generation methods share some common codes (similar use-cases of `edge_gen`. This PR solve this issue by:

+ Extracting `edge_gen`
+ Proposing a more general strategy for graph generation which is compatible with future generation engines

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,12 +6,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
 ## [Unreleased]
 ### Added
+- `pyrgg_gen_using` function added
 - Generation engine menu
 - `handle_string` function
 - `handle_pos_int` function
 - `handle_output_format` function
 - `handle_engine` function
 ### Changed
+- `dimacs_init` function removed
```
Comment (sepandhaghighi) [inline CHANGELOG.md:16]: Move to `### Remove` (Take a look at previous sections)

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@sadrasabouri Sorry for my late response 🥇 
Please take a look at my comments.

**Comment by sadrasabouri** [comment]
The newly introduced error has nothing to do with this PR and we should handle it in a separate PR.

**Comment by sepandhaghighi** [comment]
> The newly introduced error has nothing to do with this PR and we should handle it in a separate PR.

It's not a big deal.
I will take care of it at the end of the review.🔥

**Review by sepandhaghighi — APPROVED** [review]
@sadrasabouri Well done 🚀

**Pull request merged**